### PR TITLE
Core - Adding support for build data that is sent as context properties

### PR DESCRIPTION
**Pull request by gupele**

### CODE REVIEW

File: core/src/main/java/com/microsoft/applicationinsights/extensibility/initializer/BuildContextInitializer.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    final static String BUILD_DEFAULT_VALUE = "unknown";
+
+    private String gitBranch;
+    private String gitCommit;
+    private String gitUrl;
+    private boolean hasBuildData;
+
+    public BuildContextInitializer() {
+        try {
+            InputStream inputStream = getClass().getClassLoader().getResourceAsStream(BUILD_INFO_FILE_NAME);
```
Comment (amnonshmsft) [inline core/src/main/java/com/microsoft/applicationinsights/extensibility/initializer/BuildContextInitializer.java]: Please use the **[PropertyHelper](https://github.com/Microsoft/ApplicationInsights-Java/blob/master/core/src/main/java/com/microsoft/applicationinsights/internal/util/PropertyHelper.java)** class.

Also, in a separate check in please factor out the build info from the PropertyHelper class (see our **[old discussion here](https://github.com/Microsoft/ApplicationInsights-Java/pull/172)**).

File: core/src/main/java/com/microsoft/applicationinsights/internal/config/TelemetryConfigurationFactory.java [hunk 2]
```diff
@@ -141,6 +142,7 @@ private void setContextInitializers(ContextInitializersXmlElement contextInitial
         // To keep with prev version. A few will probably be moved to the configuration
         initializerList.add(new SdkVersionContextInitializer());
         initializerList.add(new DeviceInfoContextInitializer());
+        initializerList.add(new BuildContextInitializer());
```
Comment (amnonshmsft) [inline core/src/main/java/com/microsoft/applicationinsights/internal/config/TelemetryConfigurationFactory.java]: This depends on whether we plan to submit this code as part of the SDK or just as an example

**Comment by amnonshmsft** [comment]
Please do not merge before syncing with me

**Comment by littleaj** [comment]
This pull request is very old and is being pruned. The branch still exists if someone thinks this should be reopened.

**Pull request closed without merging**